docs/kubernetes-reliability.md

```diff
@@ -0,0 +1,101 @@
+Distributed system such as Kubernetes are designed to be resilient to the
+failures.  More details about Kubernetes High-Availability (HA) may be found at
+[Building High-Availability Clusters](https://kubernetes.io/docs/admin/high-availability/)
+
+To have a simple view the most of parts of HA will be skipped to describe
+Kubelet<->Controller Manager communication only.
+
+By default the normal behavior looks like:
+
+1. Kubelet updates its status to Master's API Server periodically described at
```

Comment:
Let's use consistent language and keep it less wordy:
Kubelet updates it status to apiserver periodically, as specified by `--node-status-update-frequency`